dependency_cache.go

```diff
@@ -334,12 +335,18 @@ func (d DependencyCache) downloadHttp(uri string, destination string, mods ...Re
 	defer out.Close()
 
 	if _, err := io.Copy(out, resp.Body); err != nil {
-		return fmt.Errorf("unable to copy from %s to %s\n%w", uri, destination, err)
+		return fmt.Errorf("unable to copy from %s to %s\n%w", sanitizeUri(uri), destination, err)
 	}
 
 	return nil
 }
 
+func sanitizeUri(uri string) string {
+	uriCreds := regexp.MustCompile(`:\/\/.*:.*@`)
+	uriCredsRepl := "://***:***@"
+	return uriCreds.ReplaceAllString(uri, uriCredsRepl)
+}
+
```

Comment:
I think my preference would be to parse the URI using `url.Parse` and then call the `Redacted` method on it. https://pkg.go.dev/net/url#URL.Redacted. That will redact the password. I feel like that is probably sufficient, but if you feel strongly that we should redact both the user and password, then I would suggest implementing it in a similar way.

You can parse the URL and then replace the user/password part in the parsed component, then write it out. Something like this: https://go.dev/play/p/cmdsJ7pyB28

The tricky part is that parsing can fail & how should we handle that? 

I think my thought would be to attempt to parse it. If the parse fails, then log the URL & error at *debug* level, then have the redact/sanitize method return a string like "URL has been redacted (see DEBUG log level)" and just not display any of it.

Reply:
I agree this would be a cleaner solution. It would also be in line with how the http client itself behaves when failing to fetch a URL (obfuscates password only). I will get at it soon.

Reply:
Thanks for the update!